source/extensions/common/wasm/wasm.cc

```diff
@@ -934,66 +950,66 @@ Wasm::Wasm(absl::string_view vm, absl::string_view id, absl::string_view initial
       initial_configuration_(initial_configuration) {
   wasm_vm_ = Common::Wasm::createWasmVm(vm);
   id_ = std::string(id);
-  if (wasm_vm_) {
+}
 
+void Wasm::registerFunctions() {
 #define _REGISTER(_fn) registerCallback(wasm_vm_.get(), #_fn, &_fn##Handler);
-    _REGISTER(getTotalMemory);
-    _REGISTER(_emscripten_get_heap_size);
-    _REGISTER(_llvm_trap);
+  _REGISTER(getTotalMemory);
+  _REGISTER(_emscripten_get_heap_size);
```

Comment:
Perhaps those 2 functions should be registered only if `is_emscripten_ == true`?